miner/worker.go

```diff
@@ -219,8 +219,9 @@ func newWorker(config *Config, chainConfig *params.ChainConfig, engine consensus
 	go worker.taskLoop()
 
 	// Submit first work to initialize pending state.
-	worker.startCh <- struct{}{}
-
+	if init {
+		worker.startCh <- struct{}{}
```

Comment:
Why do we need this? Couldn't we simply not init the pending block on startup?

Reply:
If so, for normal node(non-mining node), the pending state is always nil.